webcrypto/elliptic_curve.go

```diff
@@ -1,5 +1,213 @@
+// parameter into `SubtleCrypto.Sign` or `SubtleCrypto.Verify“ when using the
+// ECDSA algorithm.
+//
+// As defined in the [specification](https://www.w3.org/TR/WebCryptoAPI/#EcdsaParams-dictionary)
+type ECDSAParams struct {
+	// Name should be set to AlgorithmKindEcdsa.
+	Name AlgorithmIdentifier `json:"name"`
+
+	// Hash identifies the name of the digest algorithm to use.
+	// You can use any of the following:
+	//   * [Sha256]
+	//   * [Sha384]
+	//   * [Sha512]
+	Hash AlgorithmIdentifier `json:"hash"`
+}
+
+// EcKeyGenParams  represents the object that should be passed as the algorithm
+// parameter into `SubtleCrypto.GenerateKey`, when generating any
+// elliptic-curve-based key pair: that is, when the algorithm is identified
+// as either of AlgorithmKindEcdsa or AlgorithmKindEcdh.
+//
+// As defined in the [specification](https://www.w3.org/TR/WebCryptoAPI/#EcKeyGenParams-dictionary)
+type EcKeyGenParams struct {
+	// Algorithm holds the base algorithm description.
+	// Its name should be set to either ECDSA or ECDH.
+	Algorithm
+
+	// NamedCurve holds (a String) the name of the curve to use.
+	// You can use any of the following: CurveKindP256, CurveKindP384, or CurveKindP521.
+	NamedCurve EllipticCurveKind `json:"namedCurve"`
+}
+
+// Ensure AesKeyGenParams implements the From interface.
+var _ From[map[string]interface{}, EcKeyGenParams] = EcKeyGenParams{}
+
+// From implements the From interface for EcKeyGenParams, and initializes the
+// EcKeyGenParams instance from a map[string]interface{}.
+func (e EcKeyGenParams) From(dict map[string]interface{}) (EcKeyGenParams, error) {
+	var params EcKeyGenParams
+	nameFound := false
+	namedCurveFound := false
+
+	for key, value := range dict {
+		if strings.EqualFold(key, "name") {
+			name, ok := value.(string)
+			if !ok {
+				return EcKeyGenParams{}, NewError(0, SyntaxError, "name property should hold a string")
+			}
+
+			name = strings.ToUpper(name)
+
+			if !IsAlgorithm(name) {
+				return EcKeyGenParams{}, NewError(0, NotSupportedError, "unsupported algorithm name")
+			}
+
+			params.Name = name
+			nameFound = true
+			continue
+		}
+
+		if strings.EqualFold(key, "namedCurve") {
+			namedCurve, ok := value.(string)
+			if !ok {
+				return EcKeyGenParams{}, NewError(0, SyntaxError, "namedCurve property should hold a string")
+			}
+
+			namedCurve = strings.ToUpper(namedCurve)
+
+			if !IsEllipticCurve(namedCurve) {
+				return EcKeyGenParams{}, NewError(0, NotSupportedError, "unsupported elliptic curve name")
+			}
+
+			params.NamedCurve = EllipticCurveKind(namedCurve)
+			namedCurveFound = true
+			continue
+		}
+	}
+
+	if !nameFound {
+		return EcKeyGenParams{}, NewError(0, SyntaxError, "missing algorithm name")
+	}
+
+	if !namedCurveFound {
+		return EcKeyGenParams{}, NewError(0, SyntaxError, "missing elliptic curve name")
+	}
+
+	return params, nil
+}
+
+// Ensure AesKeyGenParams implements the From interface.
```

Comment:
That's why most comments are evil :sweat_smile:

Reply:
Hehe, I'm gonna blame Copilot here 😄 🐱 👍🏻

Reply:
I addressed it here, and in the other relevant place. Looking back at it, I find that although the comment needed to be fixed indeed, it is still valuable as a lot of Go developers don't know this syntax (I've actually seen question about it quite often) 👍🏻